### PR TITLE
feat(knock_requests_list) : implement design

### PR DESCRIPTION
**Pull request by ganfra**
## Content

Implements design of https://www.figma.com/design/7TqjqdMBaPpm3IavKsMYu6/Ask-to-join-(Knocking)?node-id=3177-33987
Data is not branched yet.
## Motivation and context
Part of https://github.com/element-hq/element-x-android/issues/3692

## Tests

Not testable yet.

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsNode.kt [hunk 1]
```diff
@@ -140,7 +145,8 @@ class RoomDetailsNode @AssistedInject constructor(
             openPollHistory = ::openPollHistory,
             openAdminSettings = this::openAdminSettings,
             onJoinCallClick = ::onJoinCall,
-            onPinnedMessagesClick = ::openPinnedMessages
+            onPinnedMessagesClick = ::openPinnedMessages,
+            onKnockRequestsClick = ::openKnockRequestsLists
```
Comment (bmarty) [inline features/roomdetails/impl/src/main/kotlin/io/element/android/features/roomdetails/impl/RoomDetailsNode.kt:149]:
```suggestion
            onKnockRequestsClick = ::openKnockRequestsLists,
```

File: features/knockrequests/impl/src/main/res/values/localazy.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <string name="screen_knock_requests_list_ban_alert_confirm_button_title">"Yes, decline and ban"</string>
+  <string name="screen_knock_requests_list_ban_alert_description">"Are you sure you want to decline and ban %1$s? This user won’t be able to request access to join this room again."</string>
+  <string name="screen_knock_requests_list_ban_alert_title">"Decline and ban from accessing"</string>
+  <string name="screen_knock_requests_list_decline_alert_confirm_button_title">"Yes, decline"</string>
+  <string name="screen_knock_requests_list_decline_alert_description">"Are you sure you want to decline %1$s request to join this room?"</string>
+  <string name="screen_knock_requests_list_decline_alert_title">"Decline access"</string>
+  <string name="screen_knock_requests_list_decline_and_ban_action_title">"Decline and ban"</string>
+  <string name="screen_knock_requests_list_empty_state_description">"When somebody will ask to join the room, you’ll be able to see their request here."</string>
+  <string name="screen_knock_requests_list_empty_state_title">"No pending request to join"</string>
```
Comment (bmarty) [inline features/knockrequests/impl/src/main/res/values/localazy.xml:15]: I think there is a typo here, `request` should have the plural form.

**Review by bmarty — APPROVED** [review]
Nice work, thanks!
Maybe a preview of the `RoomDetailsView` with some know requests could be nice

**Comment by ganfra** [comment]
> Nice work, thanks! Maybe a preview of the `RoomDetailsView` with some know requests could be nice

I'll add in another PR, because I'll need to rework a bit this screen again, to reorganise sections.

**Pull request merged**